harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -597,14 +593,14 @@ def _compute_validation_metrics(self) -> workload.Metrics:
                 num_inputs, num_batches = [sum(n) for n in zip(*input_counts)]
 
         else:
-            check.true(self._evaluate_full_dataset_defined())
+            if not self._evaluate_full_dataset_defined():
+                raise RuntimeError()
```

Comment:
use an assertion here since it's validating our own code instead of validating user code.

Reply:
Done.